sentry-core/src/main/java/io/sentry/core/SendCachedEvent.java

```diff
@@ -0,0 +1,115 @@
+package io.sentry.core;
+
+import static io.sentry.core.ILogger.logIfNotNull;
+
+import io.sentry.core.util.Objects;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.TestOnly;
+
+final class SendCachedEvent {
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+  @TestOnly static final String CACHED_EVENT_SUFFIX = ".sentry-event";
+  private final ISerializer serializer;
+  private final IHub hub;
+  private final ILogger logger;
+
+  SendCachedEvent(ISerializer serializer, IHub hub, ILogger logger) {
+    this.serializer = Objects.requireNonNull(serializer, "Serializer is required.");
+    this.hub = Objects.requireNonNull(hub, "Hub is required.");
+    this.logger = Objects.requireNonNull(logger, "Logger is required.");
+  }
+
+  public void sendCachedFiles(@NotNull File directory) {
+    if (!directory.exists()) {
+      logIfNotNull(
+          logger,
+          SentryLevel.WARNING,
+          "Directory '%s' doesn't exist. No cached events to send.",
+          directory.getAbsolutePath());
+      return;
+    }
+    if (!directory.isDirectory()) {
+      logIfNotNull(
+          logger,
+          SentryLevel.ERROR,
+          "Cache dir %s is not a directory.",
+          directory.getAbsolutePath());
+      return;
+    }
+
+    logIfNotNull(
+        logger,
+        SentryLevel.DEBUG,
+        "Processing %d items from cache dir %s",
+        directory.length(),
+        directory.getAbsolutePath());
+
+    for (File file : directory.listFiles()) {
```

Comment:
instead of the `if` below, you may use `listFiles(FilenameFilter filter)` which also returns `File[]`

Reply:
I started with the filter but thought of writing logs instead. The directory is expected to be used solely by Sentry's SDK event caching. But if it's somehow mixed up with other stuff, would be nice to know when debugging something is weird.

Reply:
that makes total sense, agreed